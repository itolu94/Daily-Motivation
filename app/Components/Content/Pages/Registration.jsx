import React, {Component} from 'react';
import API from '../../Utility/API.jsx';

export default class Registration extends Component {
    constructor(){
        super();
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            userName: '',
            email: '',
            phoneNumber: '',
        }
    }

    handleSubmit(e){
        e.preventDefault();
        let {userName, email, phoneNumber} = this.state;
        let newUser = {userName, email, phoneNumber};
        API.createAccount(newUser, (resp)=>{
            console.log(resp);
        })
    }

    handleChange(e){
        this.setState({[e.target.name]: e.target.value});
    }

    render(){
        return(
            <div id='registration-div' className="container">
                <div className="row login">
                    <div className="col m8 offset-m2 center-align">
                        <h3 id='registration-header'> Sign up to receive free motivational quotes to start your day</h3>
                        <form onSubmit={this.handleSubmit}>
                            <input
                                type="text"
                                name="userName"
                                placeholder='First Name'
                                value={this.state.userName}
                                onChange={this.handleChange}
                                required

                            />
                            <input
                                type='text'
                                name='email'
                                placeholder='Email Address'
                                value={this.state.email}
                                onChange={this.handleChange}
                                required
                            />
                            <input
                                type='text'
                                name='phoneNumber'
                                placeholder='Phone Number'
                                value={this.props.phoneNumber}
                                onChange={this.handleChange}
                                required
                            />

                            <input className='' type="submit"/>
                        </form>
                    </div>
                </div>
            </div>
        )
    }
}