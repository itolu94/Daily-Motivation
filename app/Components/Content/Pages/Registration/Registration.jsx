import React, {Component} from 'react';
import API from '../../../../Utility/API.jsx';
import {RegistrationDiv, RegistrationHeader} from "./Registration.style.jsx";

export default class Registration extends Component {
    constructor(){
        super();
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            userName: '',
            email: '',
            phoneNumber: '',
            response: ''
        }
    }

    handleSubmit(e){
        e.preventDefault();
        let {userName, email, phoneNumber} = this.state;
        let newUser = {userName, email, phoneNumber};
        API.createAccount(newUser, (response)=>{
                console.log(response);
                if(response.status === 200 || response.status === 409) this.setState({response: response.data});
                else this.setState({response: "Oops.. something went wrong.  Please try again later"});
        });
    }

    handleChange(e){
        this.setState({[e.target.name]: e.target.value});
    }

    render(){
        return(
            <RegistrationDiv className="container">
                <div className="row pages-div">
                    <div className="col m8 offset-m2 center-align">
                        <RegistrationHeader > Sign up to receive free motivational quotes to start your day</RegistrationHeader>
                        <p>{this.state.response}</p>
                        <form onSubmit={this.handleSubmit}>
                            <input
                                type="text"
                                name="userName"
                                placeholder='First Name'
                                value={this.state.userName}
                                onChange={this.handleChange}
                                required

                            />
                            <input
                                type='text'
                                name='email'
                                placeholder='Email Address'
                                value={this.state.email}
                                onChange={this.handleChange}
                                required
                            />
                            <input
                                type='text'
                                name='phoneNumber'
                                placeholder='Phone Number'
                                value={this.state.phoneNumber}
                                onChange={this.handleChange}
                                required
                            />
                            <input className='registration-submit' type="submit"/>
                        </form>
                    </div>
                </div>
            </RegistrationDiv>
        )
    }
}