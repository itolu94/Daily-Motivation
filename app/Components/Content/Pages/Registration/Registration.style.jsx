import styled from 'styled-components';

const RegistrationDiv = styled.div`
    background-color: powderblue;
    color: black;
    margin-top: 10%;
    padding: 10px 0;
`;

const RegistrationHeader = styled.h1`
    font-size: 1.92rem;
    margin:0;
    padding:0;
    height:100%;
`;

export {RegistrationDiv, RegistrationHeader};