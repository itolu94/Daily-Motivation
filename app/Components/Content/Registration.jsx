import React, {Component} from 'react';


export default class Registration extends Component {
    constructor(){
        super();
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            firstName: '',
            email: '',
            phoneNumber: '',
        }
    }

    handleSubmit(e){
        e.preventDefault();
    }

    handleChange(e){
        this.setState({[e.target.name]: e.target.value});
    }

    render(){
        return(
            <div id='registration-div' className="container">
                <div className="row login">
                    <div className="col m8 offset-m2 center-align">
                        <h3 id='registration-header'> Sign up to receive free motivational quotes to start your day</h3>
                        <form onSubmit={this.handleSubmit}>
                            <input
                                type="text"
                                name="firstName"
                                placeholder='First Name'
                                value={this.state.firstName}
                                onChange={this.handleChange}
                                required

                            />
                            <input
                                type='text'
                                name='email'
                                placeholder='Email Address'
                                value={this.state.email}
                                onChange={this.handleChange}
                                required
                            />
                            <input
                                type='text'
                                name='phoneNumber'
                                placeholder='Phone Number'
                                value={this.props.phoneNumber}
                                onChange={this.handleChange}
                                required
                            />

                            <input className='' type="submit"/>
                        </form>
                    </div>
                </div>
            </div>
        )
    }
}