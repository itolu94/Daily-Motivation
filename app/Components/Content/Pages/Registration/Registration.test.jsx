import React from 'react';
import { shallow } from 'enzyme'
import Registration from './Registration.jsx';


test("Registration component", () => {
    const wrapper = shallow(<Registration/>);
    expect(wrapper).toMatchSnapshot();
    wrapper.setState({
        userName: 'itoodud',
        email: 'djdsj@mail.com',
        phoneNumber: '9999999999',
    });
    expect(wrapper).toMatchSnapshot();
    // wrapper.find('.registration-submit').simulate('click');
    // expect(wrapper.state().response).toEqual("Account was created");
})