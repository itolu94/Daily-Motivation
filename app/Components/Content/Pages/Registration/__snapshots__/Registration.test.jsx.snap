// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component 1`] = `
<div
  className="container"
  id="registration-div"
>
  <div
    className="row pages-div"
  >
    <div
      className="col m8 offset-m2 center-align"
    >
      <h3
        id="registration-header"
      >
         Sign up to receive free motivational quotes to start your day
      </h3>
      <p />
      <form
        onSubmit={[Function]}
      >
        <input
          name="userName"
          onChange={[Function]}
          placeholder="First Name"
          required={true}
          type="text"
          value=""
        />
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          required={true}
          type="text"
          value=""
        />
        <input
          name="phoneNumber"
          onChange={[Function]}
          placeholder="Phone Number"
          required={true}
          type="text"
          value=""
        />
        <input
          className="registration-submit"
          type="submit"
        />
      </form>
    </div>
  </div>
</div>
`;

exports[`Registration component 2`] = `
<div
  className="container"
  id="registration-div"
>
  <div
    className="row pages-div"
  >
    <div
      className="col m8 offset-m2 center-align"
    >
      <h3
        id="registration-header"
      >
         Sign up to receive free motivational quotes to start your day
      </h3>
      <p />
      <form
        onSubmit={[Function]}
      >
        <input
          name="userName"
          onChange={[Function]}
          placeholder="First Name"
          required={true}
          type="text"
          value="itoodud"
        />
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          required={true}
          type="text"
          value="djdsj@mail.com"
        />
        <input
          name="phoneNumber"
          onChange={[Function]}
          placeholder="Phone Number"
          required={true}
          type="text"
          value="9999999999"
        />
        <input
          className="registration-submit"
          type="submit"
        />
      </form>
    </div>
  </div>
</div>
`;
